src/Video.tsx

```diff
@@ -249,6 +255,38 @@ const Video = forwardRef<
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [props?.paused, props.repeat, props?.showCastingOptions, props?.showControls]);
 
+  useEffect(() => {
+    const heartbeatTracker = setInterval(() => {
+      if (videoRef.current && !pausedRef.current && !seeking.current) {
+        trackVideoEvent?.('playing', Math.round(cTime.current));
+      }
+      if (youtubeId && !pausedRef.current) {
+        webViewRef.current?.injectJavaScript(`trackVideoPlaying(); true;`);
+      }
+    }, HEARTBEAT_INTERVAL);
+
+    return () => clearInterval(heartbeatTracker);
+  }, []);
+
+  useEffect(() => {
+    const completionTime = 0.95 * content.length_in_seconds;
```

Comment:
Or couldn't you just check when the progress update is made if it's around 0.95 * content.length_in_seconds then also call the track endpoint?

Reply:
@csillaj Yea I wasn't sure how to do this efficiently. It was requested that the track video complete is fired once the user has reached 95% of the video instead of the actual end, so I couldn't use the onEnd event for vimeo or youtube. 

I initially planned to use the heartbeat interval for this as well, but in some cases, the videos are too short for it to work properly. 

I can revisit this to try and find a better solution.

Reply:
@csillaj Okay I changed this to use setTimeout instead, which I think is a much better solution. It'll use other events to get the current time, and clear and update the timeout delay accordingly. I tested it and it was working well. Let me know what you think.